### PR TITLE
[Bug 15623] The machine() function returns "unknown" in LC 7.x (on OSX)

### PR DESCRIPTION
**Pull request by livecodepanos**

### CODE REVIEW

File: engine/src/dskmac.cpp [hunk 1]
```diff
@@ -4711,18 +4712,20 @@ struct MCMacDesktop: public MCSystemInterface, public MCMacSystemService
...
+		// PM-2015-07-21: [[ Bug 15623 ]] machine() returns "unknown" in OSX because of Gestalt being deprecated
+		size_t t_len = 0;
+		sysctlbyname("hw.model", NULL, &t_len, NULL, 0);
+
+		if (t_len)
+		{
+			char *t_model = (char *)malloc(t_len * sizeof(char));
+			sysctlbyname("hw.model", t_model, &t_len, NULL, 0);
+
+			return MCStringCreateWithCStringAndRelease(t_model, r_string);
```
Comment (runrevmark) [inline engine/src/dskmac.cpp]: @runrevpanos: Careful here - AndRelease semantics do not free the buffer if an error is thrown. Therefore you need to do:

```
if (!MCStringCreateWithCStringAndRelease(t_model, r_string))
{
  free(t_model);
  return false;
}
return true;
```

**Comment by runrevmark** [comment]
I should have said on the previous comment - the reason this is different is that MCMemoryNew and friends follow the engine calling convention with regard 'throwing' errors. So if MCMemoryNew fails, it 'throws' an out-of-memory error and returns false. This means that the reason for the failure can be seen in the error at the script level (assuming the memory situation isn't so terminal that the OS kills the process, anyway!)

**Comment by runrevmark** [comment]
@livecode-vulcan review ok e68fc55

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok e68fc55249edbe783541d9a20c7d6d1572f36408

**Comment by livecode-vulcan** [comment]
😎 test success e68fc55249edbe783541d9a20c7d6d1572f36408
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/591)
- try-win-x86: [success](https://vulcan.livecode.com/builders/try-win-x86/builds/651)
- try-ios: [success](https://vulcan.livecode.com/builders/try-ios/builds/540)
- try-android-armv6: [success](https://vulcan.livecode.com/builders/try-android-armv6/builds/604)
- try-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/536)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/529)

**Pull request merged**